src/index.rs

```diff
@@ -136,7 +137,7 @@ impl Index {
       let done = self.index_block(&mut wtx)?;
 
       if block % 1000 == 0 {
-        wtx.commit()?;
+        last_commit_time = Self::commit_and_log(wtx, last_commit_time, self.height()?.n()).unwrap();
```

Comment:
Can you use `?` for error handling? Also thinking that `last_commit_time` can be taking by `&mut` so it can just be updated in-place:

```suggestion
    Self::commit_and_log(wtx, &mut last_commit_time, self.height()?.n())?;
```